notifications/src/main/java/com/dojo/notifications/service/FlinkTableService.java

```diff
@@ -60,14 +65,13 @@ public Leaderboard executeSingleQuery(Set<Participant> participants, String tabl
         return env.fromCollection(tupleLeaderboard);
     }
 
-    private TreeSet<Participant> convertTableResultToList(TableResult leaderboard) {
-        Iterator<Row> rowsNew = leaderboard.collect();
+    private TreeSet<Participant> convertDataStreamToList(Iterator<Tuple3<String, String, Long>> leaderboard) {
```

Comment:
We already discussed the type needs to be from interface for better abstraction and use of polymorphism